github/with_appengine.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2017 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build appengine
+
+// This file provides glue for making github work on App Engine.
+// In order to get the entire github package to compile with
+// Go 1.6, you will need to rewrite all the import "context" lines.
+// Fortunately, this is easy with "gofmt":
+//
+//     gofmt -w -r '"context" -> "golang.org/x/net/context"' *.go
```

Comment:
Asking users to modify the source code of a package in order to get it to work is pretty unusual, and I feel this comment would be hard to spot.

I think a nice thing we can do for users is mention this in a new "App Engine" section of README (and package comment, to keep them in sync). It can simply state the current situation, that the package is expected not to build by default and needs to be modified, and point to this file for further instructions. Similar to what you wrote in https://github.com/google/go-github/pull/582#issuecomment-286822789.